src/huggingface_hub/hf_api.py

```diff
@@ -7391,6 +7397,8 @@ def create_inference_endpoint(
         account_id: Optional[str] = None,
         min_replica: int = 1,
         max_replica: int = 1,
+        scaling_metric: Optional[str | EndpointsScalingMetric] = None,
```

Comment:
`huggingface_hub` still supports Python 3.9, so we have to use `Union` instead of the pipe syntax
```suggestion
        scaling_metric: Optional[Union[str, EndpointsScalingMetric]] = None,
```